src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/transform.py

```diff
@@ -29,15 +29,46 @@ def create_transform(client, account_name, resource_group_name,
     return client.create_or_update(resource_group_name, account_name, transform_name, outputs, description)
 
 
-def add_transform_output(client, account_name, resource_group_name, transform_name, presets):
+def add_transform_output(client, account_name, resource_group_name, transform_name, preset,
+                         audio_insights_only=False, audio_language=None, on_error=None,
+                         relative_priority=None):
+    from azure.mgmt.media.models import (OnErrorType, Priority)
+
     transform = client.get(resource_group_name, account_name, transform_name)
 
-    for preset in presets:
-        transform.outputs.append(get_transform_output(preset))
+    validate_arguments(preset, audio_insights_only, audio_language)
+    transform_output = get_transform_output(preset)
+
+    if preset == 'VideoAnalyzer':
+        transform_output.preset.audio_language = audio_language
+        transform_output.preset.audio_insights_only = audio_insights_only
+    elif preset == 'AudioAnalyzer':
+        transform_output.preset.audio_language = audio_language
+
+    if on_error is not None:
+        transform_output.on_error = OnErrorType(on_error)
+
+    if relative_priority is not None:
+        transform_output.relative_priority = Priority(relative_priority)
+
+    transform.outputs.append(transform_output)
 
     return client.create_or_update(resource_group_name, account_name, transform_name, transform.outputs)
```

Comment:
This is not related to this PR but we have to change this to `client.update()` operation to avoid creating a transform by running `output add`

Reply:
Changed!